### PR TITLE
Feature fides optimizer

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: pypesto/optimize/optimizer.py [hunk 1]
```diff
@@ -790,3 +795,78 @@ def nlopt_objective(x, grad):
 
     def is_least_squares(self):
         return False
+
+
+class FidesOptimizer(Optimizer):
+    """
+    Global/Local optimization using the trust region optimizer fides.
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:802]: add a link to the github page (or rtd, when exists)?

File: pypesto/optimize/optimizer.py [hunk 2]
```diff
@@ -790,3 +795,78 @@ def nlopt_objective(x, grad):
...
+                 hessian_update: Optional[fides.HessianApproximation] =
+                 None,
+                 options: Optional[Dict] = None):
+        """
+        Parameters
+        ----------
+        options:
+            Optimizer options.
+
+        hessian_update:
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:815]: make it clearer that None means the objective's Hessian (/approximation) is used?

**Review by yannikschaelte — APPROVED** [review]

File: pypesto/optimize/optimizer.py (hunk 2 above)
Comment (FFroehlich) [reply pypesto/optimize/optimizer.py:816]: 👍

File: pypesto/optimize/optimizer.py (hunk 1 above)
Comment (FFroehlich) [reply pypesto/optimize/optimizer.py:802]: added

**Pull request merged**